### DISCUSSION TITLE
Let user select time zone

### DISCUSSION THREAD

**Discussion by kklyandev** [davetapley/helium-tax#7] [Ideas]
What time zone should be used for the transactions? Just thinking how to determine which transactions between Dec 31 and Jan 01 should be included in 2020 depending on the time zone. Would we use the time zone of the state where we live or some other time zone?

**Comment by davetapley** [comment]
The [**most recently asserted location**](https://intercom.help/heliumnetwork/en/articles/3599308-setting-your-location-with-video) is used.

The location from the hotspot is just a latitude/longitude, so [`geo-tz`](https://www.npmjs.com/package/geo-tz) is used to get a time zone (e.g. `America/Phoenix`).

Midnight Dec 31 and Jan 01 in **that time zone** are then used to filter transactions.

You'll also see in the spreadsheet that the first column (having the times) are given in ISO 8601 format, where [the UTC offset](https://en.wikipedia.org/wiki/ISO_8601#Time_offsets_from_UTC) will correspond to this time zone.
I.e. the first row may contain e.g. `2020-01-01T00:00:00-0700`, and the `-0700` tells us that this time is 7 hours behind UTC, as given by the `America/Phoenix` time zone.

---

With all that said, there is _undefined behavior_ if a hotspot **changes** time zone during 2020.
I'm not sure what the behavior should be for this, but I will at least add a warning ⚠️ when it is detected (see #8).

**Reply by kklyandev** [reply]
I was asking in general, about the best practices when doing tax reporting. Thank you for confirming that I should be using my local time zone. I'm writing my own tools to export the transactions and I was using your CSV to compare and confirm mine returns the same results and that's how I discovered that I'm using UTC and not including the few transactions that are past midnight in UTC but still before midnight in my US time zone. In my case I'm using the rewards API on the account, so I am not getting the transactions for a specific hotspot but for all hotspots linked to my account. 
As for your tool and how to handle the hotspot time zone change - I think it should be up to the user to pick a time zone before they run the report. If we assume that the user reports according to his time zone, then him having hotspots in different time zones should not produce reports in different time zones, it should all produce reports in the person's time zone.

**Reply by davetapley** [reply]
> I think it should be up to the user to pick a time zone before they run the report.

I definitely agree this would be a good feature ✅ 
I probably won't have time to do it, but I know others are working on similar tools which may support this, so I'll add links to those tools.
